tests/test_backend_gurobi.py

```diff
@@ -0,0 +1,312 @@
+import importlib
+
+import calliope.exceptions as exceptions
+import pytest  # noqa: F401
+import xarray as xr
+
+from .common.util import build_test_model as build_model
+from .common.util import check_error_or_warning
+
+if importlib.util.find_spec("gurobipy") is not None:
```

Comment:
Shouldn't we just add the solver to our dev requirements to avoid this one?

Avoiding it for the base requirements makes sense. For dev, not so much, imo.

Reply:
I tried, but it won't work. Our dependencies are currently pip and conda cross-compatible. On PyPI, this dependency is gurobipy, on Anaconda it is gurobi... So, this breaks our compatibility. I've still removed the check in the tests for gurobipy. I would prefer to alert the developer to the need to install gurobipy!

Reply:
Hmm... in the future we might want to switch to `pixi` to avoid this kind of stuff. It prefers `conda-forge` by default, but you can redirect dependencies in a case-by-case basis if needed...
I'm ok with leaving this as is for now though!